docs/components/index.md

```diff
@@ -0,0 +1,315 @@
+
+```cshtml
+<button class="btn btn-primary" @onclick(UpdateHeading)>
+    Update heading
+</button>
+```
+
+The following code calls the `CheckboxChanged` method when the checkbox is changed in the UI:
+
+```cshtml
+<input type="checkbox" class="form-check-input" 
+    id="callsMethodCheck" @onchange(CheckboxChanged)>
+```
+
+`OnInit` and `OnInitAsync` execute code after the component has been initialized:
+
+```csharp
+protected override void OnInit()
+{
+    ...
+}
+```
+
+```csharp
+protected override async Task OnInitAsync()
+{
+    ...
+}
+```
+
+`OnParametersSet` and `OnParametersSetAsync` are called when a component has received parameters from its parent and the values are assigned to properties. These methods are executed after `OnInit` during component initialization.
+
+```csharp
+protected override void OnParametersSet()
+{
+    ...
+}
+```
+
+```csharp
+protected override async Task OnParametersSetAsync()
+{
+    ...
+}
+```
+
+`SetParameters` can be overridden to execute code before parameters are set:
+
+```csharp
+public override void SetParameters(ParameterCollection parameters)
+{
+    ...
+    
+    base.SetParameters(parameters);
+}
+```
+
+`ShouldRender` can be overridden to suppress refreshing of the UI. If the implementation returns `true`, the UI is refreshed. Even if `ShouldRender` is overridden, the component is always initially rendered.
+
+```csharp
+protected override bool ShouldRender()
+{
+    var renderUI = true;
+
+    return renderUI;
+}
+```
+
+## Component disposal with IDisposable
+
+If a component implements [IDisposable](/dotnet/api/system.idisposable), the router disposes the component when the user navigates away from the component. A component can implement `IDisposable` in a C# base class that implements `BlazorComponent` (or `IComponent` directly). The following component uses `@implements IDisposable` and the `Dispose` method:
+
+```csharp
+@using System
+@implements IDisposable
+
+...
+
+@functions {
+    public void Dispose()
+    {
+        ...
+    }
+}
+```
+
+## Render child content
+
+Blazor components can render child content with an element that matches the child component's name. The following Index page renders the content of the `HeadingComponent` (*HeadingComponent.cshtml*):
+
+[!code-cshtml[](common/samples/2.x/ComponentsSample/Pages/Index.cshtml?start=1&end=11)]
+
+A child component can receive property assignments from their calling parent. In the following example, the `ParentComponent` sets the value of the `Title` property in the `ChildComponent`:
+
+*ParentComponent.cshtml*:
+
+[!code-cshtml[](common/samples/2.x/ComponentsSample/Pages/ParentComponent.cshtml?start=1&end=7)]
+
+*ChildComponent.cshtml*:
+
+[!code-cshtml[](common/samples/2.x/ComponentsSample/Pages/ChildComponent.cshtml)]
+
+The body of the Bootstrap-styled panel is provided by `ChildContent`. `ChildContent` is bound to the `<ChildComponent>` element's content in the parent component by convention.
```

Comment:
It's not bound; it just receives the content from inside the element in the parent.